source/Cosmos.System2_Plugs/System/IO/FileImpl.cs

```diff
@@ -162,11 +162,31 @@ public static void WriteAllBytes(string aFile, byte[] aBytes)
 
         public static void Copy(string srcFile, string destFile)
         {
-            using (var xFS = new FileStream(srcFile, FileMode.Open))
+            try
             {
-                var xBuff = new byte[(int)xFS.Length];
-                var yFS = new FileStream(destFile, FileMode.Create);
-                yFS.Write(xBuff, 0, xBuff.Length);
+                byte[] srcFileBytes = File.ReadAllBytes(srcFile);
+                File.WriteAllBytes(destFile, srcFileBytes);
+            }
+            catch (IOException ioEx)
+            {
+                throw new IOException("File Copy", ioEx);
+            }
+        }
+
+        public static void Copy(string srcFile, string destFile, bool overwriting)
+        {
+            if (overwriting)
```

Comment:
If it's not overwriting, and the file doesn't exist, the copy has to be done.

Reply:
Oh right ! I will modify this.